src/queue.ts

```diff
@@ -63,12 +64,16 @@ export class Queue<TItem> extends EventEmitter {
     function processItem (message : any) : PromiseLike<any> {
       let body = <TItem>JSON.parse(message.Body)
 
-      return Bluebird.resolve(body)
-        .then(handler)
+      return Bluebird.resolve()
+        .then(() => handler(body, message))
```

Comment:
Another possibility is change:
```
return Bluebird.resolve()
  .then(() => handler(body, message))
  ...
```
for
```
return Bluebird.resolve([body, message])
  .spread(handler)
  ...
```
Doing this we avoid the arrow function, which access processItem scope.
This is more a taste issue.